roofit/roofitcore/src/RooVectorDataStore.cxx

```diff
@@ -967,6 +967,8 @@ void RooVectorDataStore::cacheArgs(const RooAbsArg* owner, RooArgSet& newVarSet,
 
   // Step 2 - reorder tracked nodes
   std::sort(trackArgs.begin(), trackArgs.end(), [](RooAbsArg* left, RooAbsArg* right){
+    //LM: exclude same comparison. This avoids an issue when using sort in MacOS  versions
+    if (left != right) return false;
```

Comment:
Doesn't this return `false` almost always? Like this, the `dependsOn` is effectively deactivated.

Reply:
Maybe `return right->dependsOn(*left) && left != right`? Indeed, `a.dependsOn(a)` returns true, so this lambda implements `<=` instead of `<` as required by. `sort`.

Reply:
Ah, yes, sorry. I committed the wrong fix !

It should be if (a == b) return false;